### PR TITLE
Weapon machines

### PR DESCRIPTION
**Pull request by RKSDude**
Adds the option for players to buy weapons from weapon machines

### CODE REVIEW

**Review by BryceLao — APPROVED** [review]
LGTM

File: Assets/Scripts/PlayerScripts/WeaponsHandler.cs [hunk 1]
```diff
@@ -206,7 +207,26 @@ public void AddWeapon(GameObject weapon)
 		{
 			availableWeapons.Add(weapon);
 			weapon.GetComponent<Weapon>().Setup(_ammoInventory);
-			SelectWeapon(availableWeapons.Count - 1);
+			// Initialized selectedIndex here because using SelectWeapon causes issues with 2 weapons becoming active at once
+			int selectedIndex = 0;
+			selectedIndex = (selectedIndex + availableWeapons.Count) % availableWeapons.Count;
```
Comment (vincetiu8) [inline Assets/Scripts/PlayerScripts/WeaponsHandler.cs:212]: Shouldn't there be a -1 somewhere? Idk

File: Assets/Scripts/Weapons/WeaponPickup.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace Weapons
 {
 	public class WeaponPickup : TimedInteractable
 	{
-		private Vector3        _offset;
+		private Vector3        _offset = new Vector3(0.4f, -0.55f, 0);
```
Comment (vincetiu8) [inline Assets/Scripts/Weapons/WeaponPickup.cs:12]: Need to set this for all guns now. Should be serialized.

**Review by vincetiu8 — CHANGES_REQUESTED** [review]
I'm still getting issues with multiple weapons being out at once.

Screenshot from 2021-12-06 17-35-54

Also I get null reference errors when I try to interact with the machine.
Screenshot from 2021-12-06 17-36-43

**Review by vincetiu8 — CHANGES_REQUESTED** [review]
The shops are broken: buying an item makes the player freeze and triggers a nullreference exception.

**Review by vincetiu8 — APPROVED** [review]
LGTM but I worked on this so I'd like someone else to approve too before merging.

**Pull request merged**